src/app/sessions/reducers/index.ts

```diff
@@ -103,26 +104,49 @@ export const {
 } = fromSessions.adapter.getSelectors(getSessionsEntitiesState);
 
 export const getFullSessions = createSelector(getAllSessions, getRooms, fromJudgesIndex.getJudges, fromHearingPartIndex.getHearingParts,
-    (sessions, rooms, judges, hearingParts) => {
+    (sessions, rooms, judges, inputHearingParts) => {
         let finalSessions: SessionViewModel[];
         if (sessions === undefined) {return []}
         finalSessions = Object.keys(sessions).map(sessionKey => {
             const sessionData: Session = sessions[sessionKey];
+            const hearingParts = Object.values(inputHearingParts).filter(hearingPart => hearingPart.session === sessionData.id);
+            const allocated = calculateAllocated(hearingParts);
+
             return {
                 id: sessionData.id,
                 start: moment(sessionData.start),
                 duration: sessionData.duration,
                 room: rooms[sessionData.room],
                 person: judges[sessionData.person],
                 caseType: sessionData.caseType,
-                hearingParts: Object.values(hearingParts).filter(hearingPart => hearingPart.session === sessionData.id),
+                hearingParts: hearingParts,
                 jurisdiction: sessionData.jurisdiction,
-                version: sessionData.version
+                version: sessionData.version,
+                allocated: allocated,
+                utilization: calculateUtilized(sessionData.duration, allocated),
+                available: calculateAvailable(sessionData.duration, allocated)
             } as SessionViewModel;
         });
+
         return Object.values(finalSessions);
     });
 
+let sessionsStatsService = new SessionsStatisticsService();
```

Comment:
I know we talked about some solution here, but I'm not sure if I'm happy about this approach of instantiating a service in the reducer